### PR TITLE
DOCSP-6184: scala x509 code snippet

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**

### CODE REVIEW

File: scala/x509.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      .applyToSslSettings(
+         (builder : SslSettings.Builder)
+         => builder.enabled(true))
+      .build();
+
+   val client : MongoClient = MongoClient(settings);
+   
+   val listDatabaseNames = client.listDatabaseNames();
+
+   listDatabaseNames.foreach(println)
```
Comment (ccho-mongodb) [inline scala/x509.scala]: Nitpick: this is probably unnecessary

File: scala/x509.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+   val db : MongoDatabase = client.getDatabase("test")
+   val collection : MongoCollection[Document] = db.getCollection("stuff")
+   collection.insertOne(Document("scalatest" -> BsonString("OK"))).subscribe((results: Completed) => println(results))
+   collection.find().collect().subscribe((results: Seq[Document]) => println(s"Found: #${results.size}"))
```
Comment (ccho-mongodb) [inline scala/x509.scala]: I think the other snippets return the id of the newly-inserted document. Could we change this to the id rather than number of docs?

File: scala/build.sbt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+ThisBuild / scalaVersion := "2.12.7"
+ThisBuild / organization := "com.mongodb"
+
+lazy val hello = (project in file("."))
```
Comment (ccho-mongodb) [inline scala/build.sbt]: I think an idiomatic sbt would use "root" instead of "hello". Not a big deal to me though.

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]
Would like to retrieve/print the id of the inserted doc similar to the other snippets if possible.

**Comment by nathan-contino-mongo** [comment]
@ccho-mongodb addressed your concerns. idiomaticified & standardized our behaviour with other snippets

File: scala/x509.scala [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+         => builder.hosts(List(new ServerAddress("localmongo1", 27017)).asJava))
+      .applyToSslSettings(
+         (builder : SslSettings.Builder)
+         => builder.enabled(true))
+      .build();
+
+   val client : MongoClient = MongoClient(settings);
+
+   val db : MongoDatabase = client.getDatabase("test")
+   val collection : MongoCollection[org.mongodb.scala.bson.collection.mutable.Document] = db.getCollection("stuff")
```
Comment (ccho-mongodb) [inline scala/x509.scala:32]: Is there a namespace collision that requires you to have the full class name for `Document`? If not, you should be able to use `Document` since you imported it on line 5.

Comment (nathan-contino-mongo) [reply scala/x509.scala:32]: Both mutable and immutable Document classes have the class name "Document", just in different packages. While it's not necessary to spell this out completely here (since I only imported the mutable Document) depending on your import setup you might end up with an immutable document instead if you just use "Document". Particularly since there are snippets and might not include imports, I felt it was better to use a more verbose classname here to prevent ambiguity particularly since Scala's error messages if you DO use the immutable Document type by mistake do not lend themselves to easy debugging of the problem.

TL;DR I'd like it to stay exactly as is. This was very intentional.

**Review by ccho-mongodb — APPROVED** [review]
Other than using the full class name on ll 32 - 33, this looks good to me. Could you make sure to update them and then I'll merge your PR.

**Comment by nathan-contino-mongo** [comment]
@ccho-mongodb take a look at my comment, I don't think it should change. Ready to merge then imo.

**Comment by ccho-mongodb** [comment]
I see your point in potentially causing confusion especially if the imports are not included in the snippet. I wonder if there's a way to avoid requiring the `Document` and retrieving the result from the `Future` to avoid verbose class names as well as avoid the need for disambiguation. I'll defer to any external reviewers on the scala driver team.
